numba/unicode.py

```diff
@@ -1388,6 +1389,33 @@ def impl(s):
     return impl
 
 
+# https://github.com/python/cpython/blob/1d4b6ba19466aba0eb91c4ba01ba509acf18c723/Objects/unicodeobject.c#L11975-L12006    # noqa: E501
+@overload_method(types.UnicodeType, 'isalnum')
+def unicode_isalnum(data):
+    """Implements UnicodeType.isalnum()"""
+
+    def impl(data):
+        length = len(data)
+        if length == 0:
+            return False
+
+        if data._is_ascii:
+            for i in range(length):
+                code_point = _get_code_point(data, i)
+                if not _Py_ISALNUM(code_point):
+                    return False
+
+        for i in range(length):
+            code_point = _get_code_point(data, i)
+            if (not _PyUnicode_IsNumeric(code_point) and 
```

Comment:
This is now failing flake8 because there's trailing whitespace and the visual indent is incorrect.
https://dev.azure.com/numba/numba/_build/results?buildId=3189&view=logs&jobId=687448c2-4012-5c9e-43b3-3520c9635a30&taskId=dba03d4a-57a7-5e59-a47a-79fe530f297a&lineStart=64&lineEnd=66&colStart=1&colEnd=1

You may find it useful to `conda install flake8`, then run `flake8 numba/unicode.py` to check whether it passes locally before pushing. There are also pre-ommit hooks available, setup instructions are here:
http://numba.pydata.org/numba-doc/dev/developer/contributing.html?#coding-conventions